### ISSUE TITLE
Beatmap wedge loses resolution at high UI scale

### ISSUE THREAD

**Issue by ThatStella7922** [ppy/osu#10109]
**Describe the bug:** 
When the UI scale is set high, the beatmap wedge will become lower quality. This has been an issue for quite a while but was dismissed due to thinking that the problem was the map’s background image.
It’s actually the game and is more visible than ever with the addition of showing star count on the wedge.

**Screenshots or videos showing encountered issue:** 
![F1257370-B01E-412E-8C91-0983F0476143](https://user-images.githubusercontent.com/10524728/92629496-9dcc2880-f28b-11ea-91aa-9d1e6d61ec15.png)
At max UI scale on iPhone XR (1792x828)

![31947F3A-68BA-4A6C-B238-3B7052AFDE55](https://user-images.githubusercontent.com/10524728/92629652-e1bf2d80-f28b-11ea-8f1c-bb1b383194ea.png)
At 1X UI scale on iPhone XR (1792x828)

You can see on the beatmap wedge in the first screenshot that it’s all pixelated, including the star rating which shouldn’t be that low res.
(Ignore the empty black section on the right, I have osu scaled to not go into the notch area)

**osu!lazer version:** 
TestFlight 2020.903.0

**Logs:**
N/A for this type of issue

**Comment by smoogipoo** [comment]
Very weird, half of it has been redrawn at the correct scale meanwhile the other half hasn't. Very fitting of the name "data corruption symphony" eh?

I wouldn't be surprised if this is similar to other issues we've had around `BufferedContainer`s that I'm planning to look into.

**Comment by bdach** [comment]
It's not even half, more like bottom half quadrant (difficulty name is also badly scaled with what seems to be nearest-neighbour)

**Comment by peppy** [comment]
Can you confirm that this happens if you first set the UI scale, then completely quit the game (force quit) and check again?

**Comment by peppy** [comment]
Actually I can reproduce. It's just the upscaling algorithm being used there (which looks to be nearest-neighbour, as suggested). It probably shouldn't be.

**Comment by smoogipoo** [comment]
How were you able to reproduce? By entering song select with the high UI scale, or by changing it whilst in song select?

There shouldn't be any scaling here - we changed the wedge a while back to redraw on scale (https://github.com/ppy/osu/pull/9393), so it should be drawn at native scale.

**Comment by peppy** [comment]
Yep, entering song select at high UI scale. I was thinking it was the mag filter for the contents, but that wouldn't make sense for the text so you have a point. Must be something weird going on with how UI scale is applied.

**Comment by Joehuu** [comment]
~Song select v2 wedge no longer has a background, so closing.~

Edit: actually this affected the text components too based on the initial report. Leaving closed though because it's outdated. Not sure if still a thing with song select v2.
